### DISCUSSION TITLE
How does kitty tell the position of previous command line prompts?

### DISCUSSION THREAD

**Discussion by xuhdev** [kovidgoyal/kitty#8133] [Q&A] [answered]
I'm playing with [`--stdin-source <STDIN_SOURCE>`](https://sw.kovidgoyal.net/kitty/launch/#cmdoption-launch-stdin-source). However, `@first_cmd_output_on_screen` often got the output wrong, sometimes giving empty results. I haven't figured out the commonality among the scenarios that it fails.

Is there a document on how kitty determines the position of previous command line prompts? Perhaps I just don't understand how  `@first_cmd_output_on_screen` works.

**Comment by kovidgoyal** [comment, marked as answer]
https://sw.kovidgoyal.net/kitty/shell-integration/#notes-for-shell-developers
